Orange/widgets/unsupervised/owhierarchicalclustering.py

```diff
@@ -970,6 +974,9 @@ def set_distances(self, matrix):
 
         self.matrix = matrix
         if matrix is not None:
+            if not np.all(np.isfinite(matrix)):
+                self.Error.not_finite_distances()
+                return
```

Comment:
I may have overlooked something, but I think this is wrong: it aborts the function without reseting the items (and related models). Note that the check above sets `matrix` to `None`. In line with this, you should probably have something like:

        if matrix is not None:
            if not np.all(np.isfinite(matrix)):
                self.Error.not_finite_distances()
                matrix = None

        self.matrix = matrix
        if matrix is not None:
            self._set_items(matrix.row_items, matrix.axis)
        else:
            self._set_items(None)

The line `self.matrix = matrix` is moved from above to the place after the check for infinite values.